sdk/src/clearingHouseUser.ts

```diff
@@ -289,51 +312,55 @@ export class ClearingHouseUser {
 		return freeCollateral.gte(ZERO) ? freeCollateral : ZERO;
 	}
 
-	public getInitialMarginRequirement(): BN {
-		const postionMarginRequirement = this.getUserAccount().positions.reduce(
-			(marginRequirement, marketPosition) => {
+	/**
+	 * @returns The margin requirement of a certain type (Initial or Maintenance) in USDC. : QUOTE_PRECISION
+	 */
+	public getMarginRequirement(type: MarginCategory): BN {
+		return this.getUserAccount()
+			.positions.reduce((marginRequirement, _position) => {
+				// clone so we dont mutate the position
+				const marketPosition = this.getClonedPosition(_position);
```

Comment:
can we only clone the object if we enter the lp position block? it will make this function more efficient.

E.g. here do:
`let marketPosition = position`

and if you enter the lp if block below:
`marketPosition = this.getClonedPosition(_position);`